src/d3a/d3a_core/rq_job_handler.py

```diff
@@ -86,25 +85,13 @@ def launch_simulation_from_rq_job(scenario, settings, events, aggregator_device_
 
         config = SimulationConfig(**config_settings)
 
-        spot_market_type = settings.get('spot_market_type', None)
-        if spot_market_type is not None:
-            if spot_market_type == SpotMarketTypeEnum.ONE_SIDED.value:
-                ConstSettings.IAASettings.MARKET_TYPE = SpotMarketTypeEnum.ONE_SIDED.value
-            if spot_market_type == SpotMarketTypeEnum.TWO_SIDED_PAY_AS_BID.value:
-                ConstSettings.IAASettings.MARKET_TYPE = (
-                    SpotMarketTypeEnum.TWO_SIDED_PAY_AS_BID.value)
-                ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE = (
-                    BidOfferMatchAlgoEnum.PAY_AS_BID.value)
-            if spot_market_type == SpotMarketTypeEnum.TWO_SIDED_PAY_AS_CLEAR.value:
-                ConstSettings.IAASettings.MARKET_TYPE = (
-                    SpotMarketTypeEnum.TWO_SIDED_PAY_AS_BID.value)
-                ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE = \
-                    BidOfferMatchAlgoEnum.PAY_AS_CLEAR.value
-            if spot_market_type == SpotMarketTypeEnum.TWO_SIDED_EXTERNAL.value:
-                ConstSettings.IAASettings.MARKET_TYPE = (
-                    SpotMarketTypeEnum.TWO_SIDED_PAY_AS_BID.value)
-                ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE = \
-                    BidOfferMatchAlgoEnum.EXTERNAL.value
+        spot_market_type = settings.get("spot_market_type")
+        bid_offer_match_algo = settings.get("Bid_offer_match_algo")
```

Comment:
Shouldn't this be lowercase?
```suggestion
        bid_offer_match_algo = settings.get("bid_offer_match_algo")
```
I suppose this has not been unit-tested, and I would encourage you to do so.

Reply:
Thanks for spotting this!
I agree that the whole`launch_simulation_from_rq_job` function needs to be unit tested, however I think I'll skip this for now to push this faster because Andrea pinged me and asked for the possibility of starting the myco-external matcer from the UI, and this PR will allow the backend to support it

Reply:
Then I'll create a jira ticket for it 👍